infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -346,7 +347,18 @@ static void restartServer(boolean resetData) {
     private static void cleanShutdown() {
         log.info("CleanShutdown: Starting Cleanup.");
         shutdownServer = true;
-        activeServer.close();
+        try {
+            CorfuServerNode current = activeServer;
+            if (current != null) {
+                current.close();
```

Comment:
In the best case would be great to get rid of a volatile variable completely. Probably next time

Reply:
I'm planning to get rid of CorfuServerNode its unnecessarily complex, with that the volatile will disappear.